Added strict typings for `ARIAMixin` with literals

DeepDoge opened the pull request:
Added stricter typings for `ARIAMixin` with a fallback to also allow any other string. Also used `${bigint}` and `${number}` to type, integer and number only properties.

I wasn't sure if I should create a new `enum` type for each ARIA property, so I inlined them in the `overrideType`.

Fixes the issue: https://github.com/microsoft/TypeScript-DOM-lib-generator/issues/1835

DeepDoge commented:
@microsoft-github-policy-service agree

saschanaz commented on baselines/dom.generated.d.ts:
```diff
@@ -2404,89 +2404,89 @@ interface ANGLE_instanced_arrays {
 interface ARIAMixin {
     /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Element/ariaAtomic) */
-    ariaAtomic: string | null;
+    ariaAtomic: ("false" | "true" | (string & {})) | null;
```
Please don't use `string & {}`. Extra context: https://github.com/microsoft/TypeScript-DOM-lib-generator/pull/1467#issuecomment-1377655691

saschanaz commented on baselines/dom.generated.d.ts:
```diff
@@ -2404,89 +2404,89 @@ interface ANGLE_instanced_arrays {
     /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Element/ariaValueText) */
     ariaValueText: string | null;
-    role: string | null;
+    role: ("toolbar" | "tooltip" | "feed" | "math" | "presentation" | "none" | "note" | "application" | "article" | "cell" | "columnheader" | "definition" | "directory" | "document" | "figure" | "group" | "heading" | "img" | "list" | "listitem" | "meter" | "row" | "rowgroup" | "rowheader" | "separator" | "table" | "term" | "associationlist" | "associationlistitemkey" | "associationlistitemvalue"  ...
```
Okay, this one is huge enough that I think exporting as an enum would be better.

saschanaz commented on inputfiles/overridingTypes.jsonc:
```diff
@@ -286,6 +286,138 @@
+                        // },
+                        // "ariaLabel": {
+                        //     "overrideType": ""
+                        // },
```
Please remove commented ones.

saschanaz commented on inputfiles/overridingTypes.jsonc:
```diff
@@ -286,6 +286,138 @@
+                "properties": {
+                    "property": {
+                        "ariaColCount": {
+                            "overrideType": "`${bigint}` | \"\""
```
Does bigint have any difference than number when used here? Quickly tried but number doesn't reject huge number either.

DeepDoge replied:
`${bigint}` only allows integers, `${number}` also allows floats.
So I used `${bigint}` for integer types and `${number}` for number types.

Also I allowed empty string `""` on these. Thinking people might use empty string instead of null.
But now since I'm also removing `string & {}` from enums, making them stricter. Should I also remove the empty string `""`?

saschanaz commented on inputfiles/overridingTypes.jsonc:
```diff
@@ -286,6 +286,138 @@
+                        //     "overrideType": ""
+                        // }
+                        "role": {
+                            "overrideType": "\"toolbar\" | \"tooltip\" | \"feed\" | \"math\" | \"presentation\" | \"none\" | \"note\" | \"application\" | \"article\" | \"cell\" | \"columnheader\" | \"definition\" | \"directory\" | \"document\" | \"figure\" | \"group\" | \"heading\" | \"img\" | \"list\" | \"listitem\" | \"meter\" | \"row\" | \"rowgroup\" | \"rowheader\" | \"separator\" | \"table\" ...
```
Where did you get this list? Have you checked everything here has multiple implementation support?

saschanaz reviewed (changes requested):
This needs some unit test, take a look here: https://github.com/microsoft/TypeScript-DOM-lib-generator/blob/main/unittests/files/autocomplete.ts

DeepDoge commented:
Removed commented items. Removed `string & {}` from types, and empty string from number and integer types.
Updated roles based on [W3C](https://w3c.github.io/aria) instead of MDN, filtered out deprecated and abstract ones.
Added type for `ariaInvalid`. MDN was missing documention for `ariaInvalid`, but has one for `aria-invalid` attribute. So I missed it as well since I was looking into Element API on MDN. `ariaInvalid` has full browser support already.

DeepDoge commented:
@saschanaz Should I include roles listed at [https://www.w3.org/TR/dpub-aria-1.1](https://www.w3.org/TR/dpub-aria-1.1)?

myasonik commented:
Stumbled across this PR - seems like a great addition! @saschanaz Anything you need before this can be merged? 

@DeepDoge I wouldn't include the dpub roles as they are only a proposed recommendation and are not fully baked.
